lib/reek/smells/smell_repository.rb

```diff
@@ -9,29 +9,37 @@ module Smells
     #
     # @api private
     class SmellRepository
-      attr_reader :detectors
+      attr_reader :detectors, :typed_detectors, :configuration, :source_via, :smell_types
 
       def self.smell_types
         Reek::Smells::SmellDetector.descendants.sort_by(&:name)
       end
 
-      def initialize(source_description = nil, smell_types = self.class.smell_types)
-        @typed_detectors = nil
-        @detectors = {}
-        smell_types.each do |klass|
-          @detectors[klass] = klass.new(source_description)
+      def initialize(source_description: nil,
+                     smell_types: self.class.smell_types,
+                     configuration: Configuration::AppConfiguration.new)
+        self.source_via         = source_description
+        self.typed_detectors    = nil
+        self.detectors          = {}
+        self.configuration      = configuration
+        self.smell_types        = smell_types
+
+        initialize_detectors
+        directive = configuration.directory_directive_for_source(source_via) ||
+          configuration.default_directive
```

Comment:
>> I think we should allow merging directives, e.g., if a smell is enabled in the default directive, and configured further in a directory directive.

I disagree. Then we would have (again) a rather obscure configuration mechanism where the user often would not be entirely sure of what's happening and where we'd have to give lengthy examples of how to use that. The way it is right now, it's simple and easy
: 
- You're in directory for which there is a directive -> this gets used.
- You're not: default directive is used.

I strongly suggest to not re-introduce such a behaviour again. We had this back in the day where we allowed multiple configuration files and back then we decided that the complexity that came along with it was not necessary.
Not to mention that the assumption was (and still is for me), that no user would use this mechanism but rather have a little configuration duplication.